src/main/java/net/dries007/tfc/common/blocks/wood/TFCBookshelfBlock.java

```diff
@@ -0,0 +1,28 @@
+package net.dries007.tfc.common.blocks.wood;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.shapes.IBooleanFunction;
+import net.minecraft.util.math.shapes.ISelectionContext;
+import net.minecraft.util.math.shapes.VoxelShape;
+import net.minecraft.util.math.shapes.VoxelShapes;
+import net.minecraft.world.IBlockReader;
+
+public class TFCBookshelfBlock extends Block
```

Comment:
I'm not sure I understand, what is the purpose of this overly complicated box handling for a bookshelf? From first glance it appears to have four rotations that are or'd together... why?